### PR TITLE
ISPN-346 Fix JSON for maps

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/IPROTO-346

### CODE REVIEW

**Comment by pruivo** [comment]
@tristantarrant needs rebase.

File: core/src/main/java/org/infinispan/protostream/impl/JsonUtils.java [hunk 1]
```diff
@@ -280,6 +285,76 @@ private static void processObject(ImmutableSerializationContext ctx, JsonParser
...
+            case INT32 -> nestedWriter.writeInt32(1, Integer.parseInt(key));
+            case INT64 -> nestedWriter.writeInt64(1, Long.parseLong(key));
+            case FIXED32 -> nestedWriter.writeFixed32(1, Integer.parseInt(key));
+            case FIXED64 -> nestedWriter.writeFixed64(1, Long.parseLong(key));
+            case SINT32 -> nestedWriter.writeSInt32(1, Integer.parseInt(key));
+            case SINT64 -> nestedWriter.writeSInt64(1, Long.parseLong(key));
+            case SFIXED32 -> nestedWriter.writeSFixed32(1, Integer.parseInt(key));
+            case SFIXED64 -> nestedWriter.writeSFixed64(1, Long.parseLong(key));
+            case UINT32 -> nestedWriter.writeUInt32(1, Integer.parseInt(key));
+            case UINT64 -> nestedWriter.writeUInt64(1, Long.parseLong(key));
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/protostream/impl/JsonUtils.java:311]: isn't double/float missing?

**Review by pruivo — COMMENTED** [review]
LGTM and tested successfully with Keycloak entities (view and query). Just drop a small question :+1:

File: core/src/main/java/org/infinispan/protostream/impl/JsonUtils.java (hunk 1 above)
Comment (tristantarrant) [reply core/src/main/java/org/infinispan/protostream/impl/JsonUtils.java:311]: Those are not valid protobuf map keys. Only scalars are supported

**Comment by pruivo** [comment]
merged! thanks @tristantarrant !

**Pull request merged**